packages/experimental/surface/src/plugins/RoutesPlugin.tsx

```diff
@@ -0,0 +1,44 @@
+//
+// Copyright 2023 DXOS.org
+//
+
+import React, { Dispatch, SetStateAction } from 'react';
+import { RouteObject, useRoutes } from 'react-router';
+import { HashRouter } from 'react-router-dom';
+
+import { definePlugin, Plugin, usePluginContext } from '../framework';
+
+export type ListViewContextValue = {
+  routes: RouteObject[];
+  setRoutes: Dispatch<SetStateAction<RouteObject[]>>;
+};
+
+export type RouterPluginProvides = {
+  router: {
+    routes: () => RouteObject[];
+  };
+};
+
+export type RouterPlugin = Plugin<RouterPluginProvides>;
+
+const routerPlugins = (plugins: Plugin[]): RouterPlugin[] => {
+  return (plugins as RouterPlugin[]).filter((p) => p.provides?.router);
+};
+
+export const RoutesPlugin = definePlugin({
+  meta: {
+    id: 'RoutesPlugin'
+  },
+  provides: {
+    context: ({ children }) => <HashRouter>{children}</HashRouter>,
+    components: {
+      default: () => {
+        const { plugins } = usePluginContext();
```

Comment:
I'd like to see a fuller example re how separate plugins can cooperate to build a set of consistent routes for a given app. my intuition is that kind of coupling belongs in the app itself.